### ISSUE TITLE
Add .autodoc to list of automatically scanned extensions?

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/AutoDoc#104]
Perhaps we should autodoc to one of the two `AUTODOC_FindMatchingFiles` calls, namely the one which computes `autodoc.files`?

**Comment by sebasguts** [comment]
Good idea.

**Comment by fingolfin** [comment]
I just tried this with your patch, and realized it might not be such a good idea right now.
Or at least not in the way I imagined: because my .autodoc files are in the doc directory, and we currently don't scan that...

So we could add that to the list of dirs to be scanned... Alternatively, we could *not* add .autodoc to the list of extensions as such, but rather only add this command:
```gap
    Append( autodoc.files, AUTODOC_FindMatchingFiles(pkgdir, [ "doc" ], [ "autodoc" ]) );
```
That is, we would only look for `doc/*.autodoc` files.

But of course some people might not want that either, or might want to at least be able to turn it off... So perhaps we should not add this after all?

**Comment by sebasguts** [comment]
Having different directories to be scanned for different file extensions seems a bit weird, since one has to keep that in mind. We could make AutoDoc scan the ```doc``` dir also for ```g{,d,i}``` files, since those files will almost certainly never be there. And keeping ```.autodoc``` files in the ```doc``` directory seems natural, so I would vote for scanning the ```doc``` dir for all files :).

On the other hand, one can always add dirs (like the ```doc``` dir) via ```scan_dirs```.

**Comment by fingolfin** [comment]
Yeah, so this was mostly about the convenience of allowing me to add "doc/foo.autodoc" files and have them appear "magically" in the manual. But I am not so sure anymore I'd really want that, because I can't really control in which order they appear... or I could, if I made sure that each only contains a single chapter, and I then tell AutoDoc in which order I want those chapters...

**Comment by sebasguts** [comment]
The ```.autodoc``` files have some issues, indeed. Maybe the best way would be not to look for them but force people to add them to the list of files themselves. This makes it possible to keep the introduced chapters in order.
